### ISSUE TITLE
Is it possible to set a max number of horizontal squares?

### ISSUE THREAD

**Issue by jpgarza93** [DustinBryant/WpfDashboardControl#4]
Thanks a lot for working on this project. I am trying to achieve building a dashboard in WPF like Asana's new dashboard creator. It pretty much already works like that with the exception of restricting the number of horizontal squares and making them stretch.

Asana.com does this pretty well I think:
![image](https://user-images.githubusercontent.com/79015116/121586574-6f85a000-c9f9-11eb-9978-39b2a7be78e8.png)

**Comment by DustinBryant** [comment]
Yes, it is possible.  You'll probably want to do all that work inside DashboardHost.xaml.cs:
https://github.com/DustinBryant/WpfDashboardControl/blob/master/WpfDashboardControl/Resources/DashboardControl/DashboardHost.xaml.cs

That's the bulk of the logic incorporated with the dashboard.  You'll probably want to look where columns are being set up or appended to and put some limit in there.

**Comment by padovanl** [comment]
@jpgarza93 have you done it? Can you please tell me how you did it?

**Comment by jpgarza93** [comment]
@padovanl I did it (take a look [here](https://docs.roboticsys.com/motion-controllers/rmp-ethercat/software-tools/rapidsetup/dashboard))

I used my own approach. using **Grid** for columns and **StackPanel** for rows.

I created a hierarchy:

Dashboard
  (contains) --> Rows
    (contains) --> Columns
      (contains) --> Widgets
      
![image](https://user-images.githubusercontent.com/79015116/170285159-a4371103-13e4-4256-b825-346c8117b878.png)
![image](https://user-images.githubusercontent.com/79015116/170285980-9906825a-d4f5-4bf6-9fc6-ca376688b59e.png)

Then created predefined sizes. I got inspired by Asana's dashboard and the WordPress Divi builder tool.

I might be able to share the project but it might take me some time to extract it. Else, maybe I can help and guide you a bit.

Packages used:

- gong-wpf-dragdrop
- MvvmLightLibs
- Newtonsoft.Json
- ActiproSoftware.Views.Wpf (this is payed, but not needed)
